gemini/reasoning-engine/tutorial_alloydb_rag_agent.ipynb

```diff
@@ -0,0 +1,823 @@
+    "\n",
+    "In this example, you'll define a function that will retrieve similar documents from the vector store using semantic search.\n",
+    "\n",
+    "For improved security measures, the tool wil use IAM-based authentication to authenticate to the databases instead of using the built-in user/password authentication."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "MpUzFTt2K3Ky"
+   },
+   "outputs": [],
+   "source": [
+    "from langchain_google_alloydb_pg import AlloyDBEngine, AlloyDBVectorStore\n",
+    "from langchain_google_vertexai import ChatVertexAI, VertexAIEmbeddings\n",
+    "\n",
+    "\n",
+    "def similarity_search(query: str):\n",
+    "    \"\"\"Searches and returns movies.\n",
+    "\n",
+    "    Args:\n",
+    "      query: The user query to search for related items\n",
+    "\n",
+    "    Returns:\n",
+    "      List[Document]: A list of Documents\n",
+    "    \"\"\"\n",
+    "    engine = AlloyDBEngine.from_instance(\n",
+    "        PROJECT_ID,\n",
+    "        REGION,\n",
+    "        CLUSTER,\n",
+    "        INSTANCE,\n",
+    "        DATABASE,\n",
+    "        # Uncomment to use built-in authentication instead of IAM authentication\n",
+    "        # user=\"postgres\",\n",
+    "        # password=PASSWORD,\n",
+    "    )\n",
+    "\n",
+    "    vector_store = AlloyDBVectorStore.create_sync(\n",
+    "        engine,\n",
+    "        table_name=TABLE_NAME,\n",
+    "        embedding_service=VertexAIEmbeddings(\n",
+    "            model_name=\"textembedding-gecko@latest\", project=PROJECT_ID\n",
+    "        ),\n",
+    "    )\n",
+    "    retriever = vector_store.as_retriever()\n",
+    "    return retriever.invoke(query)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ERxxgFTcI3DC"
+   },
+   "source": [
+    "## Deploy the service\n",
+    "\n",
+    "Now that you've specified a model, tools, and reasoning for your agent and tested it out, you're ready to deploy your agent as a remote service in Vertex AI!\n",
+    "\n",
+    "Here, you'll use the LangChain agent template provided in the Vertex AI SDK for Reasoning Engine, which brings together the model, tools, and reasoning that you've built up so far."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "k2nGSr2_JWcc"
+   },
+   "outputs": [],
+   "source": [
+    "import vertexai\n",
+    "from vertexai.preview import reasoning_engines\n",
+    "\n",
+    "vertexai.init(project=PROJECT_ID, location=\"us-central1\", staging_bucket=STAGING_BUCKET)\n",
+    "\n",
+    "remote_app = reasoning_engines.ReasoningEngine.create(\n",
+    "    reasoning_engines.LangchainAgent(\n",
+    "        model=\"gemini-1.0-pro\",\n",
```

Comment:
For most use cases, we've been recommending Gemini 1.5 Flash or Pro for new users. Is there a specific reason to use 1.0?